src/fastaq_handler.cpp

```diff
@@ -45,8 +47,19 @@ void FastaqHandler::get_next()
     }
 
     ++this->num_reads_parsed;
-    this->name = this->inbuf->name.s;
-    this->read = this->inbuf->seq.s;
+    this->name = this->next_name;
+    this->read = this->next_read;
+    this->read_ahead_next_read();
+}
```

Comment:
New implementation of `get_next()`: just set `name` and `read` to what was already read previously by `read_ahead_next_read()`, and read ahead the next sequence by calling `read_ahead_next_read()` at the end